src/main/java/graphql/analysis/values/ValueVisitor.java

```diff
@@ -0,0 +1,121 @@
+package graphql.analysis.values;
+
+import graphql.PublicSpi;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputSchemaElement;
+import graphql.schema.GraphQLInputValueDefinition;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLScalarType;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A visitor callback used by {@link ValueTraverser}
+ */
+@PublicSpi
+public interface ValueVisitor {
+
+    /**
+     * This magic sentinel value indicates that a value should be removed from a list or object versus being set to null,
+     * that is the difference between a value not being present and a value being null
+     */
+    Object ABSENCE_SENTINEL = new Object() {
+        @Override
+        public String toString() {
+            return "ABSENCE_SENTINEL";
+        }
+    };
+
+    /**
+     * Represents the elements that leads to a value and type
+     */
+    interface InputElements {
+
+        /**
+         * @return then list of input schema elements that lead to an input value.
+         */
+        List<GraphQLInputSchemaElement> inputElements();
```

Comment:
why not `get`? Just curios if we have a rule about using the bean convention or not.